naga/src/front/spv/function.rs

```diff
@@ -292,278 +292,286 @@ impl<I: Iterator<Item = u32>> super::Frontend<I> {
         );
 
         if let Some(ep) = self.lookup_entry_point.remove(&fun_id) {
-            // create a wrapping function
-            let mut function = crate::Function {
-                name: Some(format!("{}_wrap", ep.name)),
-                arguments: Vec::new(),
-                result: None,
-                local_variables: Arena::new(),
-                expressions: Arena::new(),
-                named_expressions: crate::NamedExpressions::default(),
-                body: crate::Block::new(),
-            };
+            self.deferred_entry_points.push((ep, fun_id));
```

Comment:
The diff here is a lot more readable with whitespace changes hidden:
![image](https://github.com/gfx-rs/wgpu/assets/2002109/1345ac14-96af-404c-b2da-898aacb6a738)

I extracted the entry point processing from this `if let` block into a separate `process_entry_point` function and then deferred that to be called after all functions are parsed.